_posts/2019-12-18-sentry-the-guardian-of-your-quarkus-app.adoc

```diff
@@ -0,0 +1,76 @@
+---
+layout: post
+title: 'Sentry Error Tracker - The new guardian of your Quarkus application'
+date: 2019-12-18
+tags: quarkus extension logging cloud error
+author: adamevin
+---
+
+:imagesdir: /assets/images/posts/quarkus-sentry
+
+*If like me you have met Quarkus and fell in love*, you started implementing your backend and you started to feel so light and quick, click clack, whoot whoot... *in a woosh, your lightning fast native container is ready to be sent on your brand new Kubernetes cluster 🤙*
+
+*This is awesome!!*
+
+Mmmeahh *you may have forgotten a detail...* have you thought about *the fate of your first production exception* (please be honest with me :-)?
+
+["upperalpha"]
+. As a backend god, master of the world, first of his name, you implemented some Logstash process to extract your log files, send them to some Elasticsearch server somewhere and... sorry to say it, but you just spend more time configuring it than actually coding your app 😅.
+. You thought you could manually check the error logs a few times a week and ... oops, unlucky you, your application broke exactly the day you forgot to check, let's hope no one else noticed 🤭?
+. *You were waiting for this article 🤓*
+
+Now that you are teased, aren't you? Let's dive in. 
+
+== Sentry, the guardian has arrived on Quarkus
+
+Once your code is reviewed, tested in many ways, containerized, ..., and finally *promoted to production*, [line-through]#I guess you can expect it to run smoothly ever after..# *an exception is going to pop*, it's just a question of time. The questions are:
+
+. *How will you get notified about this error?*
+. *What are you going to do about it?*
+
+Well, this looks like a complex problem right? This is what I really like about using *Quarkus and Sentry together*, you will be all set in a few minutes:
+
+* Create a https://sentry.io/signup/[Sentry account], you can choose a free plan or another depending on your project needs (or with a bit more time, even a https://docs.sentry.io/server/[self hosted sentry])
+* Configure Sentry in your Quarkus application
+. Add the extension:
++
+```
+$ ./mvnw quarkus:add-extension -Dextensions=sentry
+```
+. Set it up: 
++
+```
+quarkus.log.sentry=true
+quarkus.log.sentry.dsn=[your Sentry DSN]
+quarkus.log.sentry.in-app-packages=org.acme
+```
++ 
+NOTE: More details about the configuration can be found in the https://quarkus.io/guides/logging-sentry[Logging with Sentry guide]
```

Comment:
we shouldn't post this before that link will work.

Reply:
also would be good if https://github.com/quarkusio/quarkus/pull/6302 could be included as the doc in master states sentry.io is opensource.